### PR TITLE
Feat auth 01

### LINKED ISSUE #1: Feat: "Auth"
This issue will add the auth features to the already created routing in the app.

### PR DESCRIPTION
**Pull request by vaibhav-mahalle**
Created basic setup and added the following functionalities.

> 
> 1. Login and Signup pages
> 2. All the pages for the video library
> 3. Public and private routes for the app
> 4. authentication

### CODE REVIEW

File: src/routes/RouterPath.jsx [hunk 1]
```diff
@@ -22,6 +23,40 @@ export const RouterPath = () => {
...
+            <History />
+          </RequiresAuth>
+        }
+      />
+      <Route
+        path="/playlist"
+        element={
+          <RequiresAuth>
+            <PlayList />
+          </RequiresAuth>
```
Comment (rahulcs754) [inline src/routes/RouterPath.jsx:49]: nice to use requiresAuth

**Review by rahulcs754 — APPROVED** [review]
The code looks good and clean

File: src/Pages/Auth/Login.jsx [hunk 2]
```diff
@@ -1,9 +1,84 @@
...
+    password: "Vaibhav123",
+  };
+
+  const setLogin = () => {
+    setLoginData(testCredentials);
+  };
+
+  const handleLogin = async (e) => {
+    e.preventDefault();
+    // console.log("getting called")
```
Comment (sadathshariff) [inline src/Pages/Auth/Login.jsx:33]: remove the commented code

**Review by sadathshariff — APPROVED** [review]
Looks good, go ahead and merge it

File: src/context/Auth/util.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    if (res.status === 201) {
+      authDispatch({
+        type: "SIGNUP",
+        payload: {
+          user: res.data.foundUser,
+          token: res.data.encodedToken,
+        },
+      });
+      ToastMsg("Successfully Signed Up", "success");
+      navigate("../login");
```
Comment (RahulSingh9131) [inline src/context/Auth/util.js:45]: i think here you need to navigate to "/login" after successfully getting the token.

**Review by RahulSingh9131 — APPROVED** [review]
everything seems good to me. PR is ready to merge👍👍

File: src/Pages/Auth/Login.jsx (hunk 2 above)
Comment (vaibhav-mahalle) [reply src/Pages/Auth/Login.jsx:33]: thank you for the suggestion will make the necessary changes.

File: src/context/Auth/util.js (hunk 3 above)
Comment (vaibhav-mahalle) [reply src/context/Auth/util.js:45]: it is redirecting to log in still, I will look into it.

**Pull request merged**